processing/src/main/java/org/apache/druid/segment/DimensionIndexer.java

```diff
@@ -127,7 +127,7 @@
    * @return An array containing an encoded representation of the input row value.
    */
   @Nullable
-  EncodedKeyComponentType processRowValsToUnsortedEncodedKeyComponent(@Nullable Object dimValues, boolean reportParseExceptions);
+  EncodedDimensionValue<EncodedKeyComponentType> processRowValsToUnsortedEncodedKeyComponent(@Nullable Object dimValues, boolean reportParseExceptions);
```

Comment:
nit: javadoc params and return type needs updated.. but this isn't really new, it basically needed updated long ago, it is not always an array 😅

Reply:
Fixed.